src/api/EscargotPublic.cpp

```diff
@@ -1847,6 +1852,15 @@ bool DebuggerC::getWaitBeforeExitClient()
     return false;
 }
 
+void DebuggerC::deleteClient()
+{
+    if (m_debuggerClient) {
+        // delete DebuggerClient that was created and delivered from the user
+        delete m_debuggerClient;
+        m_debuggerClient = nullptr;
+    }
```

Comment:
`m_debuggerClient` is a non-GC-object created by the user, so it should be deleted as well